File: frontend/types/assignment.ts
export type AssignmentStatus = "Draft" | "Published";

export interface Assignment {
  id: number;
  title: string;
  description: string;
  deadline: string; // ISO date string
  maxMarks: number;
  status: AssignmentStatus;
  classId: number;
  className: string;
  classGrade: number;
  classSection: string;
  subjectId: number;
  subjectName: string;
  teacherId: number;
  teacherName: string;
  createdAt: string;
  updatedAt: string | null;
  attachmentFileName: string | null;
}

export interface AssignmentInput {
  title: string;
  description: string;
  deadline: string;
  maxMarks: number;
  classGrade: number;
  classSection: string;
  subjectId: number;
  status?: AssignmentStatus;
  attachment?: File | null;
  removeAttachment?: boolean;
}

File: backend/SchoolMS.Data/Entities/Assignment.cs
using SchoolMS.Data.Enums;

namespace SchoolMS.Data.Entities;

public class Assignment
{
    public int Id { get; set; }
    public string Title { get; set; } = string.Empty;
    public string Description { get; set; } = string.Empty;
    public DateTime Deadline { get; set; }
    public int MaxMarks { get; set; }
    public AssignmentStatus Status { get; set; } = AssignmentStatus.Draft;

    // Optional single attachment (Word/PDF/Excel/image). Null when no file was uploaded.
    public string? AttachmentFileName { get; set; }
    public string? AttachmentStoredName { get; set; }
    public string? AttachmentContentType { get; set; }
    public long? AttachmentSize { get; set; }

    public int ClassId { get; set; }
    public Class Class { get; set; } = null!;

    public int SubjectId { get; set; }
    public Subject Subject { get; set; } = null!;

    public int TeacherId { get; set; }
    public User Teacher { get; set; } = null!;

    public DateTime CreatedAt { get; set; }
    public DateTime? UpdatedAt { get; set; }

    public ICollection<Submission> Submissions { get; set; } = new List<Submission>();
}
